qwandaq/src/main/java/life/genny/qwandaq/utils/FilterUtils.java

```diff
@@ -0,0 +1,412 @@
+package life.genny.qwandaq.utils;
+
+import static life.genny.qwandaq.attribute.Attribute.PRI_CODE;
+
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
+import life.genny.qwandaq.Question;
+import life.genny.qwandaq.attribute.Attribute;
+import life.genny.qwandaq.constants.FilterConst;
+import life.genny.qwandaq.datatype.DataType;
+import life.genny.qwandaq.entity.search.trait.*;
+import life.genny.qwandaq.models.SavedSearch;
+import org.jboss.logging.Logger;
+
+import life.genny.qwandaq.Ask;
+import life.genny.qwandaq.attribute.EntityAttribute;
+import life.genny.qwandaq.entity.BaseEntity;
+import life.genny.qwandaq.entity.search.SearchEntity;
+import life.genny.qwandaq.entity.search.clause.Or;
+import life.genny.qwandaq.message.QDataBaseEntityMessage;
+import life.genny.qwandaq.models.UserToken;
+import org.w3c.dom.Attr;
+
+@ApplicationScoped
+public class FilterUtils {
+    private static final Logger log = Logger.getLogger(MethodHandles.lookup().lookupClass());
+    static Jsonb jsonb = JsonbBuilder.create();
+
+    @Inject
+    QwandaUtils qwandaUtils;
+
+    @Inject
+    BaseEntityUtils beUtils;
+
+    @Inject
+    SearchUtils searchUtils;
+
+    @Inject
+    UserToken userToken;
+
+    /**
+     * Strip search base entity code without jti
+     *
+     * @param orgSbe Original search base entity code
+     * @return Search base entity code without jti
+     */
+    public String getCleanSBECode(String orgSbe) {
```

Comment:
This method should be done differently, currently it is only looking at delimiters and not the actual JTI.

Something like this would be more appropriate:

```java
    public String getCleanSBECode(String code) {

		String jti = userToken.getJTI().toUpperCase();
		if (code.contains(jti))
			return StringUtils.removeEnd(code, jti);

		return code;
    }
```